main/src/ca/uwaterloo/flix/language/phase/Codegen.scala

```diff
@@ -531,48 +529,44 @@ object Codegen {
       visitor.visitMethodInsn(INVOKEVIRTUAL, asm.Type.getInternalName(clazz), method.getName, asm.Type.getMethodDescriptor(method), false)
       compileUnbox(ctx, visitor)(tpe)
 
-    case Expression.GetTupleIndex(base, offset, tpe, _) =>
-      // Load the Value singleton object and base expression, to call `Value.cast2tuple`, to get the elements array.
-      Constants.loadValueObject(visitor)
+    case Expression.Index(base, offset, tpe, _) =>
+      // Emit code for the base (tuple) expression.
       compileExpression(ctx, visitor, entryPoint)(base)
-      visitor.visitMethodInsn(INVOKEVIRTUAL, Constants.valueObject, "cast2tuple", "(Ljava/lang/Object;)[Ljava/lang/Object;", false)
 
-      // Now compile the offset and load the array element. Unbox if necessary.
+      // A tuple is represented as an array of objects.
+      visitor.visitTypeInsn(CHECKCAST, "[Ljava/lang/Object;")
```

Comment:
Instead of hard coding `"[Ljava/lang/Object;"`, use a combination of `asm.Type.getDescriptor` and `Constants.arrayObjectClass`.